### PR TITLE
More field work on quizzes

### PR DESCRIPTION
**Pull request by deadlybutter**
### What does this PR do?
This pr does more work around the quiz fields and specifically does a bunch of work around adding answers to the questions.

### Any background context you want to provide?
I did some refactoring to make working with the json a bit safer. Also added ID's in the php side of things like previously suggested. 

Also, there is a new `award` variable. This is what we'll use to determine what we assign points to when people answer questions.

Here is the new json structure on the `questions` property in contentful 

```
{
    "items": [
        {
            "title": "test question",
            "answers": [
                {
                    "title": "this is a title",
					"award": "winner"
                },
                {
                    "title": "this is a second title",
					"award": "loser"
                }
            ]
        },
        {
            "title": "test second question",
            "answers": [
                {
                    "title": "this is a title",
					"award": "winner"
                },
                {
                    "title": "this is a second title",
					"award": "loser"
                },
                {
                    "title": "this is a third title",
					"award": "loser"
                }
            ]
        }
    ]
}
```

### CODE REVIEW

File: app/Entities/Quiz.php [hunk 1]
```diff
@@ -7,6 +7,47 @@
...
+     * @return array
+     */
+    private function parseAnswersFromQuestion($question)
+    {
+        if (! isset($question['answers'])) {
+            return [];
+        }
+
+        return collect($question['answers'])->map(function ($answer) {
+            $data = copyFieldIfSet(['title', 'award'], $answer);
```
Comment (DFurnes) [inline app/Entities/Quiz.php]: Can we use Laravel's [`array_only`](https://laravel.com/docs/5.4/helpers#method-array-only) for this?

File: app/Entities/Quiz.php [hunk 2]
```diff
@@ -7,6 +7,47 @@
...
+            $data['id'] = uniqid();
```
Comment (DFurnes) [inline app/Entities/Quiz.php]: Could we use the array offset for this? It sounds like `uniqid()` [might not guarantee a unique value](http://php.net/manual/en/function.uniqid.php).

File: resources/assets/components/Quiz/Answer.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Answer = ({ answer }) => (
+  <div className="answer">
+    <p>{ answer.title }</p>
```
Comment (DFurnes) [inline resources/assets/components/Quiz/Answer.js]: Let's pass an explicit prop for this, not an object.

File: app/Entities/Quiz.php (hunk 2 above)
Comment (deadlybutter) [inline app/Entities/Quiz.php]: Sure!

File: app/Entities/Quiz.php (hunk 1 above)
Comment (deadlybutter) [inline app/Entities/Quiz.php]: Of course Laravel already had a helper for that 😎

File: resources/assets/components/Quiz/Question.js [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
 import PropTypes from 'prop-types';
+import Answer from './Answer';
 
-const Question = ({ question }) => (
+const Question = ({ title, answers }) => (
   <div className="question">
-    <h2>{ question.title }</h2>
+    <h2>{ title }</h2>
+    {answers.map(answer => (
+      <Answer key={answer.id} answer={...answer} />
```
Comment (DFurnes) [inline resources/assets/components/Quiz/Question.js]: Oh, one thing: remove the `answer=` so this is just `{...answer}`.

**Pull request merged**